lib/golomb/golomb.js

```diff
@@ -0,0 +1,309 @@
+    const values = [];
+
+    for (const item of items) {
+      const hash = sipmod64(item, key, this.m);
+      values.push(hash);
+    }
+
+    values.sort(compare);
+
+    let last2 = values[0];
+    let i = 1;
+
+    for (;;) {
+      const cmp = last1.cmp(last2);
+
+      if (cmp === 0)
+        break;
+
+      if (cmp > 0) {
+        if (i < values.length) {
+          last2 = values[i];
+          i += 1;
+          continue;
+        }
+        return false;
+      }
+
+      const value = this.readU64(br);
+
+      if (value === EOF)
+        return false;
+
+      last1.iadd(value);
+    }
+
+    return true;
+  }
+
+  readU64(br) {
+    try {
+      return this._readU64(br);
+    } catch (e) {
+      if (e.message === 'EOF')
+        return EOF;
+      throw e;
+    }
+  }
+
+  _readU64(br) {
+    const num = new U64(0);
+
+    // Unary
+    while (br.readBit())
+      num.iaddn(1);
+
+    const rem = br.readBits64(this.p);
+
+    return num.ishln(this.p).ior(rem);
+  }
+
+  toBytes() {
+    return this.data;
+  }
+
+  toNBytes() {
+    const bw = bio.write();
+    bw.writeVarint(this.n);
+    bw.writeBytes(this.data);
+    return bw.render();
+  }
+
+  toPBytes() {
+    const data = Buffer.allocUnsafe(1 + this.data.length);
+    data.writeUInt8(this.p, 0);
+    this.data.copy(data, 1);
+    return data;
+  }
+
+  toNPBytes() {
+    const data = Buffer.allocUnsafe(5 + this.data.length);
+    data.writeUInt32BE(this.n, 0);
+    data.writeUInt8(this.p, 4);
+    this.data.copy(data, 5);
+    return data;
+  }
+
+  toRaw() {
+    assert(this.p === 19);
+    return this.toNBytes();
+  }
+
+  fromItems(P, key, items) {
+    assert(typeof P === 'number' && isFinite(P));
+    assert(P >= 0 && P <= 32);
+    items = new BufferSet(items);
+
+    assert(Buffer.isBuffer(key));
+    assert(key.length === 16);
+
+    assert(items.size >= 0);
+    assert(items.size <= 0xffffffff);
+
+    this.p = P;
+    this.n = items.size;
+    this.m = M.mul(new U64(this.n));
+
+    const values = [];
+
+    for (const item of items) {
+      assert(Buffer.isBuffer(item));
+      const hash = sipmod64(item, key, this.m);
+      values.push(hash);
+    }
+
+    values.sort(compare);
+
+    const bw = new BitWriter();
+
+    let last = new U64(0);
+
+    for (const hash of values) {
+      const rem = hash.sub(last).imaskn(this.p);
+      const value = hash.sub(last).isub(rem).ishrn(this.p);
+
+      last = hash;
+
+      // Unary
+      while (!value.isZero()) {
+        bw.writeBit(1);
+        value.isubn(1);
+      }
+      bw.writeBit(0);
+
+      bw.writeBits64(rem, this.p);
+    }
+
+    this.data = bw.render();
+
+    return this;
+  }
+
+  fromBytes(N, P, data) {
+    assert(typeof N === 'number' && isFinite(N));
+    assert(typeof P === 'number' && isFinite(P));
+    assert(P >= 0 && P <= 32);
+    assert(Buffer.isBuffer(data));
+
+    this.n = N;
+    this.p = P;
+    this.m = M.mul(new U64(this.n));
+    this.data = data;
+
+    return this;
+  }
+
```

Comment:
This file could definitely use some docstrings to explain the methods

Reply:
Added docstrings. Added a README with a brief overview.